src/models/thread.ts

```diff
@@ -281,7 +306,7 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
                  */
                 this.replayEvents?.push(event);
             } else {
-                this.addEventToTimeline(event, toStartOfTimeline);
+                this.insertEventIntoTimeline(event);
```

Comment:
This would still be called if MSC 3981 is enabled and we get a reaction via sync. We should probably check for MSC 3981 here and just append it if MSC 3981 is enabled.

Reply:
Yes, I will do that.